framework/src/modules/random/module.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { LiskValidationError, validator } from '@liskhq/lisk-validator';
+import { BlockGenerateContext } from '../../node/generator';
+import {
+	BlockAfterExecuteContext,
+	BlockVerifyContext,
+	GenesisBlockExecuteContext,
+} from '../../node/state_machine';
+import { BaseModule, ModuleInitArgs } from '../base_module';
+import { RandomAPI } from './api';
+import { MAX_LENGTH_REVEALS_MAINCHAIN, MODULE_ID_RANDOM } from './constants';
+import { RandomEndpoint } from './endpoint';
+import { randomModuleConfig } from './schemas';
+import { ModuleConfig } from './types';
+
+export class RandomModule extends BaseModule {
+	public id = MODULE_ID_RANDOM;
+	public name = 'random';
+	public api = new RandomAPI(this.id);
+	public endpoint = new RandomEndpoint(this.id);
+
+	private _maxLengthRevealMainchain!: number;
+
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async init(args: ModuleInitArgs): Promise<void> {
+		const { moduleConfig } = args;
+		const errors = validator.validate(randomModuleConfig, moduleConfig);
+		if (errors.length) {
+			throw new LiskValidationError(errors);
+		}
+		this._maxLengthRevealMainchain =
+			(moduleConfig as ModuleConfig).maxLengthRevealsMainchain ?? MAX_LENGTH_REVEALS_MAINCHAIN;
```

Comment:
In other modules we set `moduleConfig` to `this._moduleConfig` for a reason I don't remember but it's the convention. And if you set it that way you don't have to type `(moduleConfig as ModuleConfig).configName` every time you want to use the config, instead you can do `this._moduleConfig = moduleConfig as ModuleConfig` and use it without casting. Also this way hooks can use the config too, otherwise config is trapped inside the init, though I don't know if hooks use the config.

Reply:
I think either way is fine. There are no strict rule to save whole config or just one value. which every make sense per module should be good IMO.
for example, if there are too many property in config, maybe keep it as config while if there is one, just keep the one value

Reply:
I saw both approaches in the new modules, maybe either of the approaches is fine, @mehmetegemen btw the config is not trapped in the init, you assign the properties of a config to the class properties so it can be used in other places too. To be consistent i'll use `this._moduleConfig = moduleConfig as ModuleConfig`